relay-spans/src/otel_to_sentry.rs

```diff
@@ -884,5 +945,30 @@ mod tests {
             get_path!(annotated_span.links[0].sampled),
             Some(&Annotated::new(true))
         );
```

Comment:
All the tests already have explicit assertions that are technically redundant with the snapshots, but it's especially egregious with this one. Should we just remove these?